sfe/sfe.go

```diff
@@ -0,0 +1,301 @@
+	if incomingJWT != "" {
+		claims, err := sfe.validateUnpauseJWTforAccount(incomingJWT)
+		if err != nil {
+			unpauseFailure(response)
+		}
+
+		// TODO(#7356) Declare a registration ID variable to populate an
+		// rapb unpause account request message.
+		_, innerErr := strconv.ParseInt(claims.Subject, 10, 64)
+		if innerErr != nil {
+			unpauseInvalidRequest(response)
+		}
+
+		// TODO(#7536) Send gRPC request to the RA informing it to unpause
+		// the account specified in the claim. At this point we should wait
+		// for the RA to process the request before returning to the client,
+		// just in case the request fails.
+
+		// Success, the account has been unpaused.
+		unpauseSuccessful(response, claims.Subject)
+	} else {
+		unpauseInvalidRequest(response)
+	}
+}
+
+// unpauseInvalidRequest is a helper that displays a page indicating there's no
+// action for a Subscriber to take due to an invalid or lack of JWT in the data
+// object.
+func unpauseInvalidRequest(response http.ResponseWriter) {
+	renderTemplate(response, "unpause-noParams.html", nil)
+}
+
+// unpauseSuccessful is a helper that displays a success message to the
+// Subscriber if their account has been unpaused.
+func unpauseSuccessful(response http.ResponseWriter, accountID string) {
+	type tmplData struct {
+		ShouldUnpause bool
+		AccountID     string
+	}
+	renderTemplate(response, "unpause-post.html", tmplData{true, accountID})
+}
+
+// unpauseFailure is a helper that displays a failure message to the Subscriber
+// indicating that their account has failed to unpause. This should only be used
+// when the JWT fails to validate and other failures should use
+// unpauseInvalidRequest.
+func unpauseFailure(response http.ResponseWriter) {
+	type tmplData struct {
+		ShouldUnpause bool
+	}
+	renderTemplate(response, "unpause-post.html", tmplData{false})
+}
+
+// unpausePrepareToUnpause is a helper that displays a page containing a filled
+// out form containing a JWT created by the WFE. Upon accessing that link, the
+// Subscriber would click the "unpause my account" button to submit the form
+// back to the SFE.
+func unpausePrepareToUnpause(response http.ResponseWriter, unpausePath string, incomingJWT string) {
+	type tmplData struct {
+		UnpausePath string
+		JWT         string
+	}
+	renderTemplate(response, "unpause-params.html", tmplData{unpausePath, incomingJWT})
+}
+
+// Unpause allows a requester to unpause their account via a form present on the
+// page.
+func (sfe *SelfServiceFrontEndImpl) Unpause(response http.ResponseWriter, request *http.Request) {
+	switch request.Method {
+	case http.MethodHead:
+		return
+	case http.MethodGet:
+		sfe.getHelper(response, unpauseJWT(request.URL.Query().Get("jwt")))
+	case http.MethodPost:
+		sfe.postHelper(response, unpauseJWT(request.URL.Query().Get("jwt")))
+	default:
+		response.Header().Set("Access-Control-Allow-Methods", "GET, HEAD, POST")
+		response.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
```

Comment:
In HTTP the usual dichotomy is that GET is idemopotent and POST is not necessarily. So for the actual unpausing we should allow only POST.

HEAD is defined as the same thing GET would return, but with no body. So we should simply return a 405 Method Not Allowed with no body for all non-POST requests and that will cover both GET and HEAD.

After writing this I realized I was confused: There is a single path (`/sfe/v1/unpause`), and if you GET it it returns HTML; if you POST it it unpauses. I think this might be a little confusing. I think you actually want three paths:

 - `/sfe/v1/unpause` - GET, returns an HTML informational page with a button.
 - `/sfe/v1/do-unpause` - POST, returns a redirect to `/sfe/v1/unpause-successful` on success.
 - `/sfe/v1/unpause-successful` - GET, returns an HTML informational page indicating success.
 
This [POST/redirect/GET](https://en.wikipedia.org/wiki/Post/Redirect/Get) pattern is common for a number of reasons. If the POST serves up an HTML page directly, then reloading that page will resubmit the POST. We don't want that. Once the user has successfully unpaused, they don't need to resubmit.